.github/workflows/update-pulp-packages.yml

```diff
@@ -0,0 +1,79 @@
+name: Update package after dependabot merge
+
+on:
+  push:
+    paths:
+      - automation/requirements.txt
+
+env:
+  GIT_AUTHOR_NAME: Foreman Packaging Automation
+  GIT_AUTHOR_EMAIL: packaging@theforeman.org
+  PACKAGE_TO_UPDATE: requirements.txt
+
+jobs:
+  bump-package:
+    name: "Bump packages after dependabot merge"
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.head_ref == 'rpm/develop'
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Install dependencies
+      run: |
+        sudo apt-get install -y --no-install-recommends python3-rpm rpm git-annex python3-semver
+        sudo curl --create-dirs -o /usr/local/bin/spectool https://pagure.io/rpmdevtools/raw/26a8abc746fba9c0b32eb899b96c92841a37855a/f/spectool.in
+        sudo curl --create-dirs -o /usr/local/bin/rpmdev-bumpspec https://pagure.io/rpmdevtools/raw/6f387c1deaa5cbed770310e288abde04b17421dc/f/rpmdev-bumpspec
+        sudo curl --create-dirs -o /usr/local/bin/rpmdev-vercmp https://pagure.io/rpmdevtools/raw/79740e6f1881e399b0b4340a8090dd5adc91a4ea/f/rpmdev-vercmp
+        printf '#!/bin/bash\necho "$GIT_AUTHOR_NAME <$GIT_AUTHOR_EMAIL>"' | sudo tee /usr/local/bin/rpmdev-packager
+        sudo chmod +x /usr/local/bin/*
+    - name: Sort Updated Libs
+      if: github.event_name == 'push' && github.head_ref == 'rpm/develop'
+      run: git diff --unified=0 ${{github.event.before}} ${{github.sha}} automation/requirements.txt | grep -Po '(?<=^\+)(?!\+\+).*' > $PACKAGE_TO_UPDATE
+    - name: Check if package can be updated
+      if: github.event_name == 'push' && github.head_ref == 'rpm/develop'
+      run: |
+        set +e
```

Comment:
I'd strongly advise against big shell scripts in YAML. Create a separate shell script and run that. If it's limited to just GHA, store it in `.github`. That way you can modify it with proper syntax highlighting and tools like shellcheck.